### PR TITLE
Add initial version of the realm metadata tool

### PR DESCRIPTION
**Pull request by p-sawicki2**

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    Ok(decoded)
+}
+
+pub fn arr_to_hex(bytes: &[u8]) -> String {
+    bytes.iter().fold(String::new(), |mut output, b| {
+        let _ = write!(output, "{b:02X}");
+        output
+    })
+}
```
Comment (L0czek) [inline src/utils.rs]: The hex crate you include in this project has an encode function which does exactly what the arr_to_hex functions intends to do.

File: src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+pub fn arr_to_string(arr: &[u8]) -> String {
+    let len = arr.iter().position(|&x| x == 0).unwrap_or(arr.len());
+    let valid_bytes = &arr[..len];
+    String::from_utf8(valid_bytes.to_vec()).expect("utf8 conversion error")
```
Comment (L0czek) [inline src/utils.rs:31]: Do you intend to convert a c-string (null terminated array) to a rust string? If so take a look at this https://doc.rust-lang.org/std/ffi/struct.CString.html

File: src/cli.rs
Comment (L0czek) [inline src/cli.rs:1]: How do you feel about adding default values to the parameters that align with the instruction specified in the README?

**Review by L0czek — APPROVED** [review]
It appears to be working correctly as advertised in the README.

File: src/utils.rs (hunk 1 above)
Comment (p-sawicki2) [reply src/utils.rs]: done

File: src/utils.rs (hunk 2 above)
Comment (p-sawicki2) [reply src/utils.rs:31]: The string might be not null-terminated (the case where the whole slice is filled up with characters). Also, the CString require unsafe code blocks, so I would prefer to leave this function as it is.

File: src/cli.rs
Comment (p-sawicki2) [reply src/cli.rs:1]: done

**Pull request merged**